python3Packages.word2vec: fix build

wkral opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
ZHF: #97479
Testing changed to pytest, and requires an external dataset so check phase is omitted.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

wkral committed:
python3Packages.word2vec: fix build and tests

wkral commented:
Result of `nixpkgs-review pr 98143` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.word2vec</li>
    <li>python38Packages.word2vec</li>
  </ul>
</details>

risicle commented on pkgs/development/python-modules/word2vec/default.nix:
```diff
@@ -19,12 +20,12 @@ buildPythonPackage rec {
-    ${python.interpreter} test_word2vec.py
-  '';
+  # Checks require test data that needs to be downloaded separately
+  doCheck = false;
```
If you're going to disable the tests, could you at least add a `pythonImportsCheck` so we have _some_ hope of detecting breakage?

(also, is there no subset of the tests which _doesn't_ use the test dataset which you could enable?)

wkral replied:
Thanks I didn't know about `pythonImportCheck`.

The only test that was passing ironically was an import check so it didn't seem worth it. However, I tried downloading the support data to see if I could get that to work and it did. My worry with this version, is that hydra may impose some heavy bandwidth costs on a host that I'm unsure is equiped to handle them. Though, it's obviously better for ensuring the package works.

If this version looks good to you I'll squash the commits and push that up. Otherwise I'll just add the `pythonImportCheck`.

risicle replied:
How big _is_ the support data?

wkral replied:
It's around 30 MiB, but the tests only use a small portion of that, notice the head command. The host is a personal website not a company or institution, which is why I hesitate. I expect the package devepers only really expect people to be downloading it infrequently when they setup the project on their dev machines for the first time.

Do you have some idea how often Hydra would end up downloading it? I don't think it maintains a cache for stuff like that.

Maybe I could just check in the test data file? It's only 100KB zipped it's 33KB. I suppose there are licencing implications with that though.

risicle replied:
I think hydra does keep a cache of these files.

risicle replied:
And a 33K test file would almost definitely be a no-go. We're even trying to keep patch files out of the repo these days.

risicle commented:
This package helpfully has `gcc` hardcoded into its build system, so to get this working (again?) on macos, you'll need to provide `pkgs.gcc` in the `nativeBuildInputs`. That having been said, when I do so, I still get 3 test failures on macos 10.14.

wkral commented:
No problem, I'll add gcc, looks like it's in the python code, so yeah can't just override it with make flags. 

> I still get 3 test failures on macos 10.14.

Which 3 are those? Maybe I can figure out why.

risicle commented:
https://gist.github.com/risicle/a34a7c1e6105b52fdb7dd819ce6c38dd

It's _possible_ that these are only failing because the mac I have access to is managed and has a few random annoying restrictions added by the IT team 🙄 .

wkral commented:
The last two failures are a result of the first failure with the `-6` return code. Which apparently means the `word2vec` sub process aborted. Could be a memory allocation issue, but I'm not too sure.

wkral commented:
Result of `nixpkgs-review pr 98143` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python37Packages.word2vec</li>
    <li>python38Packages.word2vec</li>
  </ul>
</details>

wkral commented:
It's not clear to me when it last worked on macOS, the latest build on `trunk` show it hitting the missing `gcc` issue. So I've fixed that, and if there is a cache for the intermeidate build inputs then I doubt the bandwidth will be too bad. So I squashed all the commits.

Happy to improve things if there is another suggestion on how to make it work on macOS, but maybe it would even work on one of the hydra builders given your potential IT issues.

jonringer reviewed (approved):
LGTM

Result of nixpkgs-review pr 98143 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python37Packages.word2vec</li>    <li>python38Packages.word2vec</li>  </ul></details>

risicle commented:
Yup, a step in the right direction :+1:
